Fix man proxy in FreeBSD

tesuji opened the pull request:
In FreeBSD, `man` command only [searches for man files in MANPATH][1]. 
Which means that FreeBSD `man` program does not accept interpret local man files.

On Bash-compatible shell, one could use this command to open rust specific manpages:

```sh
env MANPATH=$HOME/.rustup/toolchains/stable-x86_64-unknown-freebsd/share/man man rustc
```

[1]: https://www.freebsd.org/cgi/man.cgi?query=man&apropos=0&sektion=0&manpath=FreeBSD+12.0-RELEASE+and+Ports&arch=default&format=html

Locally tested on:
- [x] Ubuntu
- [x] FreeBSD 11

r? @kinnison 

**Note:** Ping me if someone needs to squash this PR.

tesuji commented on src/cli/rustup_mode.rs:
```diff
@@ -1025,16 +1025,17 @@ fn doc(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
+    let mut manpaths = toolchain.path().to_path_buf();
+    manpaths.push("share");
+    manpaths.push("man");
+    utils::assert_is_directory(&manpaths)?;
```
Is it necessary to check for existence of the man dir?
Can we assume that Rustup always install manpages in this location?

kinnison replied:
I think it's not unreasonable to fail more informatively if someone types `rustup man` and the manpages moved, the error might be more helpful than a simply missing or incorrect manpage

kinnison reviewed (commented):
I think this looks good, though I'm loathe to approve because `-M` overrides the `MANPATH` environment variable which might also be set and relevant.  I'd be more comfortable with manipulating that than passing `-M`

tesuji commented:
@kinnison So you want to manipulate `MANPATH` env var via [`Command::env`](https://doc.rust-lang.org/std/process/struct.Command.html#method.env)?

kinnison commented:
I figure that if `MANPATH` is set, we have to do that to be correct (i.e. prepend our path onto `MANPATH`)   If it's not set, we could use `-M` but at that point the code is simpler to always prepend to `MANPATH`

tesuji commented:
@kinnison I pushed the change you requested.

kinnison reviewed (commented):
This looks reasonable to me.  Have you verified it against FreeBSD and at least one Linux?

kinnison commented:
If so, please rebase to clean up the history of the branch before I merge.

tesuji committed:
Fix man proxy in FreeBSD

In FreeBSD, `man` command only searches for man files in MANPATH [1].

[1]: https://www.freebsd.org/cgi/man.cgi?query=man&apropos=0&sektion=0&manpath=FreeBSD+12.0-RELEASE+and+Ports&arch=default&format=html

tesuji commented:
@kinnison I tested on Ubuntu. FreeBSD I only test the equivalent shell command.

kinnison reviewed (approved):
LGTM
